src/test/java/BaseTest.java

```diff
@@ -1,11 +1,177 @@
+        WebElement avatarIcon = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("img[class='avatar']")));
+        avatarIcon.click();
+    }
+
+    public String generateRandomName(){
+        return UUID.randomUUID().toString().replace("-","");
+    }
+
+    public void provideCurrentPassword(String password) {
+        WebElement currentPassword = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[name='current_password']")));
+        currentPassword.clear();
+        currentPassword.sendKeys(password);
+
+    }
+
+    public void provideProfileName(String newProfileName){
+        WebElement profileName = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[id='inputProfileName']")));
+        profileName.clear();
+        profileName.sendKeys(newProfileName);
+    }
+
+    public void clickOnSaveBtn(){
+        WebElement saveBtn = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[class='btn-submit']")));
+        saveBtn.click();
+    }
+  //Mouse Hover
+    public WebElement clickPlayBtn() {
+        WebElement playButton = driver.findElement(By.cssSelector("[data-testid='play-btn']"));
+        actions.moveToElement(playButton).perform();
+        return wait.until(ExpectedConditions.visibilityOf(playButton));
+
+    }
+
+    public void clickDeletePlaylistBtn() {
+        WebElement deletePlaylistBtn = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button.del.btn-delete-playlist")));
+        deletePlaylistBtn.click();
+    }
+
+    public void clickAnyPlaylist() {
+        WebElement clickPlaylist = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".playlist:nth-child(3)")));
+        clickPlaylist.click();
+    }
+
+    public void registrationLinkClick() {
+        WebElement registrationLink = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[href='registration']")));
+        registrationLink.click();
+    }
+
+    public boolean isSongPlaying() {
+            WebElement soundBar = wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//div[@data-testid='sound-bar-play']")));
+            return soundBar.isDisplayed();
+        }
+
+    public void enterNewPlaylistName() {
+        WebElement inputField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='name']")));
+        inputField.sendKeys(Keys.chord(Keys.CONTROL, "A", Keys.BACK_SPACE));
+        inputField.sendKeys(newPlaylistName);
+        inputField.sendKeys(Keys.ENTER);
+    }
+
+    public void doubleClickPlaylist() {
+        WebElement dblClick = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".playlist:nth-child(3)")));
+        actions.doubleClick(dblClick).perform();
+    }
+
+    public String getRenamePlaylistSuccessMsg(){
+   WebElement notification = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.success.show")));
+   return notification.getText();
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.

**BaseTest** — Implements the necessary methods for work with webdriver.

The BaseTest class contains common setup and teardown methods, as well as any shared functionality or configurations required by multiple test classes.

**BaseTest**

The problem we are trying to solve here is the reuse of common behaviors across test classes avoiding coding duplication and centralizing these actions at one point.

**@BeforeSuite**	- The annotated method will be run before all tests in this suite have run.

**@BeforeMethod** -	The annotated method will be run before all the test methods in the current class have been run.

**@AfterMethod**	- The annotated method will be run after each test method.

_Example without a Base Test Class_

![image](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/c3abad8a-8bc1-4ad0-8a66-23bd19e7af48)